src/libguac/client.c

```diff
@@ -185,6 +185,12 @@ static void guac_client_promote_pending_users(union sigval data) {
     /* Acquire the lock for reading and modifying the list of pending users */
     guac_rwlock_acquire_write_lock(&(client->__pending_users_lock));
 
+    /* Skip user promotion entirely if there's no pending users */
+    if (client->__pending_users == NULL) {
+        guac_rwlock_release_lock(&(client->__pending_users_lock));
+        goto promotion_complete;
```

Comment:
To be more DRY, should the same be done for:

https://github.com/apache/guacamole-server/blob/974816a5caae45f5ecde5b4f2b258b9a513f2c06/src/libguac/client.c#L200-L205

Reply:
Good call - it's done.